index.js

```diff
@@ -0,0 +1,50 @@
+const container = document.querySelector('.container');
+const addBtn = document.querySelector('#add');
+let bookCollection = JSON.parse(localStorage.getItem('bookCollection')) || [];
+
+// Function to remove a book
+function removeBook(book) {
+  bookCollection = bookCollection.filter((b) => b !== book);
+  localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
+}
+
+// Function to display all books in the collection
+function displayBooks() {
+  container.innerHTML = '';
+  bookCollection.forEach((book) => {
+    const newBook = document.createElement('div');
+    newBook.innerHTML = `<h2>${book.title}</h2><p>by ${book.author}</p><button class="remove">Remove</button><hr>`;
+    container.appendChild(newBook);
+
+    // Adding event listener to the "Remove" button
+    const removeBtn = newBook.querySelector('.remove');
+    removeBtn.addEventListener('click', () => {
+      removeBook(book);
+      newBook.remove();
+    });
+  });
+}
+
+// Function to add a new book
+function addBook(title, author) {
+  const newBook = { title, author };
+  bookCollection.push(newBook);
+  localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
+  displayBooks();
+}
```

Comment:
- [OPTIONAL] I highly recommend you always provide the unique identification `id` to your newly added book. It will help you while filtering your array by removing the correct book at a time.

- [OPTIONAL] I highly recommend you avoid using some methods which can mutate your array by using destructuring method while adding a new book. Read [this lesson](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) for more info.


```suggestion
function addBook(title, author) {
  const newBook = { title, author, id: Date.now() };
  bookCollection = [...bookCollection, newBook]
  localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
  displayBooks();
};
```